libs/langchain/langchain/tools/github/utils.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Optional
+
+from github import GithubException
+from github.Issue import Issue
+
+from langchain.chat_models import ChatOpenAI
+from langchain.chat_models.base import BaseChatModel
+from langchain.output_parsers import StrOutputParser
+from langchain.prompts.chat import (
+    ChatPromptTemplate,
+    HumanMessagePromptTemplate,
+    SystemMessagePromptTemplate,
+)
+
+if TYPE_CHECKING:
+    from github.Repository import Repository
+
+
+def generate_branch_name(issue: Issue, llm: Optional[BaseChatModel] = None) -> str:
```

Comment:
@KastanDay is this function used? If not, might be worth deleting this file and having it in your local env, or adding it to a template that demonstrates how to use some of these Github tools!

Reply:
No it's not, my thought was that others might find it useful to call directly. Helpful util for Github agents. 

Up to you if worth including, deletion is fine.

Reply:
Or happy to demo it in the`github.ipynb` docs.

Reply:
I'll delete for now. Feels a bit niche for the main lib, and would love to see it in a template some day! Happy to help you construct a github template if you're game at any point. Would be a really cool use case to demo!

https://github.com/langchain-ai/langchain/blob/master/templates/docs/CONTRIBUTING.md